### PR TITLE
switch id and name in app-profile

### PR DESCRIPTION
**Pull request by muraliQlogic**
Partial fix for #137  (it's a good idea to open an issue first for discussion)

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: test/app-profile.js [hunk 1]
```diff
@@ -81,16 +81,16 @@ describe('Bigtable/AppProfile', function() {
     });
 
     it('should expand name into full resource path', function() {
-      assert.strictEqual(appProfile.id, APP_PROFILE_ID);
+      assert.strictEqual(appProfile.name, APP_PROFILE_ID);
```
Comment (sduskis) [inline test/app-profile.js]: Please change `APP_PROFILE_ID` to `APP_PROFILE_NAME`

File: test/app-profile.js [hunk 2]
```diff
@@ -155,7 +155,7 @@ describe('Bigtable/AppProfile', function() {
         options_,
         callback
       ) {
-        assert.strictEqual(name, appProfile.name);
+        assert.strictEqual(name, appProfile.id);
```
Comment (sduskis) [inline test/app-profile.js]: please change the `name` field to be `id`

Comment (muraliQlogic) [reply test/app-profile.js]: @sduskis  :Changes committed as per comments.

**Comment by sduskis** [comment]
App Profiles did not exist in previous versions, otherwise, this would be a breaking change.

File: test/instance.js [hunk 3]
```diff
@@ -773,9 +773,9 @@ describe('Bigtable/Instance', function() {
 
       instance.getAppProfiles(function(err, appProfiles, apiResponse) {
```
Comment (sduskis) [inline test/instance.js]: Please also change

```
   var APP_PROFILE_NAME = 'my-app-profile';
```

to 
```
   var APP_PROFILE_ID = 'my-app-profile';
```

and also update any usages of that variable.

File: src/app-profile.js [hunk 4]
```diff
@@ -35,18 +35,16 @@ var Cluster = require('./cluster.js');
...
  * const appProfile = instance.appProfile('my-app-profile');
  */
-function AppProfile(instance, name) {
+function AppProfile(instance, id) {
   this.bigtable = instance.bigtable;
   this.instance = instance;
 
-  var id = name;
-
   if (id.indexOf('/') === -1) {
```
[the remark this answers is not in the document]
Comment (kolodny) [reply src/app-profile.js]: I don't have a strong opinion either way. However if we do keep it, I would suggest we compare the first bit to the current value of `instance.id` to make sure the user didn't mix up accounts or something, like `assert(this.id.includes(instance.id))`

[the remark this answers is not in the document]
Comment (sduskis) [reply src/app-profile.js]: That check sounds reasonable.

File: system-test/bigtable.js [hunk 5]
```diff
@@ -200,7 +200,7 @@ describe('Bigtable', function() {
       APP_PROFILE.getMetadata(function(err, metadata) {
         assert.strictEqual(
           metadata.name,
-          APP_PROFILE.id.replace('{{projectId}}', bigtable.projectId)
```
Comment (sduskis) [inline system-test/bigtable.js]: Please also change `APP_PROFILE_NAME` to `APP_PROFILE_ID` in this file.

**Comment by sduskis** [comment]
@muraliQlogic, can you please resolve the conflicts?

**Comment by sduskis** [comment]
@muraliQlogic, can you please resolve the conflicts?

**Comment by sduskis** [comment]
@muraliQlogic, can you please add a comment saying "I signed it!"

**Comment by muraliQlogic** [comment]
I signed it!

**Pull request merged**